crates/biome_js_analyze/src/lint/style/no_parameter_assign.rs

```diff
@@ -54,6 +59,35 @@ declare_lint_rule! {
     /// }
     /// ```
     ///
+    /// ## Options
+    ///
+    /// The `noParameterAssign` rule can be configured using the `propertyAssignment` option, which determines whether property assignments on function parameters are allowed or denied. By default, `propertyAssignment` is set to `allow`.
+    ///
+    /// ```json
+    /// {
+    ///   "noParameterAssign": {
+    ///     "propertyAssignment": "allow"
+    ///   }
+    /// }
+    /// ```
```

Comment:
Few suggestions: 
- Move this block after `### propertyAssignment`. This scales better, because in case we add a new option in the future, we don't need to update this text
- As the code snippet, what I meant is to add a `json` code snippet with `deny` (modify this one), and then add a code snippet with `js,expect_diagnostic,use_options`. This way we show a diagnostic and example to the user of when `deny` is used.